src/js/components/List/stories/RenderedList.js

```diff
@@ -22,16 +22,8 @@ export const RenderedList = () => (
   <Box align="center" pad="large">
     <List
       data={data.slice(0, 10)}
-      primaryKey={(item) => (
-        <Text key={item.entry} size="large" weight="bold">
-          {item.entry}
-        </Text>
-      )}
-      secondaryKey={(item) => (
-        <Text key={item.location} size="small" color="dark-4">
-          {item.location}
-        </Text>
-      )}
+      primaryKey={(item) => item.entry}
+      secondaryKey={(item) => item.location}
```

Comment:
Can we preserve the `Text` component being passed to the primary and secondaryKey properties along with the styling?

Reply:
@jcfilben yes we can preserve it but instead of doing  
```
 primaryKey={(item) => (
        <Text key={item.entry} size="large" weight="bold">
          {item.entry}
        </Text>
 )}
```  
we can do
 ```
primaryKey={(item) => [
        item.entry,
        <Text size="large" weight="bold">
          {item.entry}
        </Text>,
 ]}
```  
so that it takes the returning value of the getValue function as an object instead of number or string in 👇
        https://github.com/grommet/grommet/blob/420cecb3148948c9efbe3635b51926324135b9f0/src/js/components/List/List.js#L364-L374
        
And then we can change the above conditional render as
```
           else if (primaryKey) {
                  const primary = getValue(item, index, primaryKey);
                  content =
                    typeof primary === 'string' ||
                    typeof primary === 'number' ? (
                      <Text key="p" weight="bold">
                        {primary}
                      </Text>
                    ) : (
                      <Text key="p" {...primary[1].props}>
                        {primary[0]}
                      </Text>
                    );
```
This approach allows us to preserve the ` Text ` component along with its styling. Additionally, I've observed that we don't need to modify how we pass the ` secondaryKey `. The warning in the console is specifically related to the ` primaryKey `, so we can leave the ` secondaryKey ` unchanged.